src/super.c

```diff
@@ -344,11 +354,32 @@ int scoutfs_read_super(struct super_block *sb,
 		goto out;
 	}
 
-	*super_res = *super;
 	ret = 0;
 out:
-	kfree(super);
-	return ret;
+	if (ret < 0) {
+		kfree(super);
+		return ERR_PTR(ret);
+	} else
+		return super;
+}
+
+/*
+ * Read the super block from meta dev.
+ */
+int scoutfs_read_super(struct super_block *sb,
+		       struct scoutfs_super_block *super_res)
+{
+	struct scoutfs_sb_info *sbi = SCOUTFS_SB(sb);
+	struct scoutfs_super_block *super_block;
+
+	super_block = scoutfs_read_super_from_bdev(sb, sbi->meta_bdev);
+	if (IS_ERR(super_block))
```

Comment:
This whole dance makes me think that we should have consistency between read_super() and read_super_from_bdev().  Either always fill the caller's or return allocated.  I think I'd lean towards filling the caller's because then they can re-use a long-lived buffer across multiple read_super calls as part of some loop?  Like the quorum central loop.